### ISSUE TITLE
How to get all objects if the User has the global permission?

### ISSUE THREAD

**Issue by humitos** [django-guardian/django-guardian#248]
I'm trying to use "get_objects_for_user" in a User that has the global permission and I'm getting the empy list instead all the instances of that model.

For example, 
 1. I created a permission called "core.edit_event" on "core" app and "Event" model
 1. I created some instances of Event model
 1. then I assigned this permission globally (not per object) to User "test"
```
>>> from guardian.shortcuts import assign_perm
>>> u = User.objects.get(username='test')
>>> assign_perm('core.edit_event', u)
<Permission: core | event | Edit event>
```
 1. then I called "guardian.shortcuts.get_objects_for_user" on "core.edit_event" and "test" User and I get the Empty list instead all of the instances of Event model.
```
>>> from guardian.shortcuts import get_objects_for_user
>>> get_objects_for_user(u, 'core.edit_event')
[]
```
I mean, if the user has the global permissions "core.edit_event" it should return all the instances of that model. Do I misunderstanding the concept?

**Comment by v-erena** [comment]
Last week I came across django-guardian and ran into the same issue desribed here. I was actually expecting the same as you, humitos. I want to list all instances of a model in a class based view for which the user has view permissions, either set as object based permission or as global permission. 
I guess, this behaviour is useful for others as well? 
humitos, did you solve this already for your use case?

**Comment by lukaszb** [comment]
That's a good point. Going to implement that by adding an `use_global_perms` param unless someone wants to work on it. @v-erena, interested in some karma points?

**Comment by v-erena** [comment]
I was already diving a bit into the code of the function get_objects_for_user() and added the following under superuser and anonymous user check:
```python
    #...
    # First check if user is superuser and if so, return queryset immediately
    if with_superuser and user.is_superuser:
        return queryset

    # Check if the user is anonymous. The
    # django.contrib.auth.models.AnonymousUser object doesn't work for queries
    # and it's nice to be able to pass in request.user blindly.
    if user.is_anonymous():
        user = get_anonymous_user()

    #check for global permissions directly based on perms input parameter
    has_global_permissions = all(user.has_perm(perm) for perm in perms)
    if has_global_permissions:
        return queryset
    #...

```
As I understand from code above, perms should not be used as errors can arise due to mixed content types etc. I do not yet understand why. I am pretty new to Django... but maybe this is a good starting point?
For me, my test cases in my project turn 'green' now :)

**Comment by humitos** [comment]
Hi. Sorry, I'm not working in that project anymore and I think we didn't fix that.

**Comment by v-erena** [comment]
Thanks for your quick answer. I already forked and work on tests.

**Comment by v-erena** [comment]
Ok, I realize that my approach shown above really was too naive, as I did not take into account the flags any_perm and use_groups for instance. @lukaszb: I guess I need your help to fully understand what's going on in that function. Furthermore, the function get_objects_for_group also needs global permission checks. These two functions are quite similar, and it looks to me that code is duplicated for the initial checks. We could refactor this a bit. How to proceed here now? Shall I try my best and then go for a pull request?

**Comment by aidanlister** [comment]
Yeah, this really tripped me up. Is there a fix in the pipeline?

**Comment by v-erena** [comment]
there is a pull request already filed: 
https://github.com/lukaszb/django-guardian/pull/293

**Comment by brianmay** [comment]
Is this issue resolved now that #293 has been merged?

**Comment by brianmay** [comment]
No response; assuming it is fixed.
